pyccl/halos/halo_model.py

```diff
@@ -23,83 +23,69 @@ class HMCalculator(CCLAutoreprObject):
     an arbitrary function of mass, scale factor and Fourier scales.
 
     Args:
-        cosmo (:class:`~pyccl.core.Cosmology`): a Cosmology object.
-        massfunc (:class:`~pyccl.halos.hmfunc.MassFunc`): a mass
-            function object.
-        hbias (:class:`~pyccl.halos.hbias.HaloBias`): a halo bias
-            object.
-        mass_def (:class:`~pyccl.halos.massdef.MassDef`): a mass
-            definition object.
-        log10M_min (float): logarithmic mass (in units of solar mass)
-            corresponding to the lower bound of the integrals in
-            mass. Default: 8.
-        log10M_max (float): logarithmic mass (in units of solar mass)
-            corresponding to the upper bound of the integrals in
-            mass. Default: 16.
-        nlog10M (int): number of samples in log(Mass) to be used in
-            the mass integrals. Default: 128.
+        mass_function (str or :class:`~pyccl.halos.hmfunc.MassFunc`):
+            the mass function to use
+        halo_bias (str or :class:`~pyccl.halos.hbias.HaloBias`):
+            the halo bias function to use
+        mass_def (str or :class:`~pyccl.halos.massdef.MassDef`):
+            the halo mass definition to use
+        lM_min, lM_max (float): lower and upper integration bounds
+            of logarithmic (base-10) mass (in units of solar mass).
+            Default range: 8, 16.
+        nlM (int): number of uniformly-spaced samples in log(Mass)
+            to be used in the mass integrals. Default: 128.
         integration_method_M (string): integration method to use
             in the mass integrals. Options: "simpson" and "spline".
             Default: "simpson".
-        k_min (float): some of the integrals solved by this class
+        k_norm (float): some of the integrals solved by this class
```

Comment:
I am not convinced this is the way to deal with normalised profiles. More later.

Reply:
You mean the name? In `star` I proposed `k_large_scale`.

Reply:
See my global comment about normprof. Will resolve this for now.

Reply:
I am unresolving this - we have to decide what to do now with that parameter. It doesn't belong in `hmc` any more.

Reply:
Yes. We don't need this parameter and should be deprecated

Reply:
done